label_studio/projects/serializers.py

```diff
@@ -166,6 +166,34 @@ def validate_label_config(self, value):
             self.instance.validate_config(value)
         return value
 
+    def validate_model_version(self, value):
+        """Custom model_version validation"""
+        p = self.instance
+
+        # Only run the validation if model_version is about to change
+        # and it contains a string
+        if p is not None and p.model_version != value and value != '':
+            # that model_version should either match live ml backend
+            # or match version in predictions
+
+            if (
+                p.ml_backends.filter(title=value).exists()
+                or p.predictions.filter(project=p, model_version=value).exists()
```

Comment:
I think you could do (pseudocode)

```
MlBackend.objects.filter(project=p, title=value).union(Prediction.objects.filter(project=p, model_version=value)).exists()
```

this would be one query instead of four, assuming p.ml_backends and p.predictions aren't prefetched.